### ISSUE TITLE
[Bug] Solidity type alias yields `unresolved custom type:`

### ISSUE THREAD

**Issue by gpsanant** [alloy-rs/core#744]
### Component

sol! macro

### What version of Alloy are you on?

v0.8.3

### Operating System

macOS (Apple Silicon)

### Describe the bug

The following abi 

[DelegationManager.json](https://github.com/user-attachments/files/17109326/DelegationManager.json)

along with the following code

```rust
mod core {
    alloy::sol!(
        #[allow(missing_docs)]
        #[sol(rpc)]
        DelegationManager,
        "../abis/DelegationManager.json"
    );
}
```

```rust
error: unresolved custom type: DelegatedShares
  --> scenario-generator/src/bin/populate_src.rs:63:5
   |
63 | /     alloy::sol!(
64 | |         #[allow(missing_docs)]
65 | |         #[sol(rpc)]
66 | |         DelegationManager,
67 | |         "../abis/DelegationManager.json"
68 | |     );
   | |_____^
   |
   = note: this error originates in the macro `$crate::sol_types::sol` which comes from the expansion of the macro `alloy::sol` (in Nightly builds, run with -Z macro-backtrace for more info)
```

If you prefer to build the contracts yourself:
```
git clone git@github.com:Layr-Labs/eigenlayer-contracts.git
git checkout slashing-magnitues
forge b -C src/contracts
```

See `out/DelegationManager.sol/DelegationManager.json`.

I'm using a type alias `DelegatedShares` for `uint256`.

**Comment by mattsse** [comment]
yeah this looks like an issue with type aliases

```json
         {
            "name": "delegatedShares",
            "type": "uint256[]",
            "internalType": "DelegatedShares[]"
          }
```

**Comment by DaniPopes** [comment]
There's 2 issues here:
1. We're ignoring UDVT arrays, fixed in https://github.com/alloy-rs/core/pull/745
2. After this fix, it still fails because the type doesn't have an assigned contract because it is declared globally, this requires a bit more work to fix

**Comment by gpsanant** [comment]
Thanks for the prompt response.

For now, I'm actually not using the functions with the custom types so I'll just redefine interfaces.

If I included the types inside of the library contract in which they're defined, would that address (2)?

**Comment by DaniPopes** [comment]
Yes I think so, if you were to move the `type DelegatedShares is uint256;` inside of one of the interfaces

**Comment by gpsanant** [comment]
Cool, confirming that
1. getting rid of the alias array
2. stuffing the alias definition inside of the library

Made the contract get imported correctly.

Unfortunately, that makes the solidity syntax quite gross, so I'm temporarily going with
```bash
sed -i '' 's/DelegatedShares/uint256/g' out/DelegationManager.sol/DelegationManager.json
```
which seems to work? :)

**Comment by dobbobalina2** [comment]
@DaniPopes  Any solutions for using the entrypoint.sol contract from the account abstraction lib found [here](https://github.com/eth-infinitism/account-abstraction/blob/6f02f5a28a20e804d0410b4b5b570dd4b076dcf9/contracts/core/EntryPoint.sol)?

Importing PackedUserOperation is an issue

**Comment by kayabaNerve** [comment]
I believe I have a similar issue here when using the `sol` macro. I defined an interface which inherits and used types from its parent. When not using the `sol` macro, yet my own expander which outputs to a file (I need a preprocessor per #602), I get a distinct error of:

> proc-macro-error2 API cannot be used outside of `entry_point` invocation, perhaps you forgot to annotate your #[proc_macro] function with #[proc_macro_error]

I'm unsure exactly how that became my error yet wanted to log it here.
